[MessageInputView] Sometimes crashes when a GIF is tapped from keyboard

Choosing any gif/sticker file from keyboard causes the crash.

Device: Samsung Galaxy A50 using Samsung Keyboard

Here's the crash log:

```
java.lang.RuntimeException: java.lang.reflect.InvocationTargetException
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:506)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100)

     Caused by: java.lang.reflect.InvocationTargetException
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100) 

     Caused by: java.io.FileNotFoundException: /data/user/0/com.appname.chat/cache/STREAM_114831131/image/gif.gif: open failed: ENOENT (No such file or directory)
        at libcore.io.IoBridge.open(IoBridge.java:496)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:235)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:186)
        at com.getstream.sdk.chat.utils.StorageHelper.getCachedFileFromUri(StorageHelper.kt:41)
        at com.getstream.sdk.chat.view.messageinput.MessageInputController.sendNormalMessage(MessageInputController.kt:100)
        at com.getstream.sdk.chat.view.messageinput.MessageInputController.onSendMessageClick$stream_chat_android_release(MessageInputController.kt:87)
        at com.getstream.sdk.chat.view.messageinput.MessageInputView.onSendMessage(MessageInputView.kt:281)
        at com.getstream.sdk.chat.view.messageinput.MessageInputView.sendGifFromKeyboard(MessageInputView.kt:350)
        at com.getstream.sdk.chat.view.messageinput.MessageInputView.access$sendGifFromKeyboard(MessageInputView.kt:47)
        at com.getstream.sdk.chat.view.messageinput.MessageInputView$configInputEditText$3.onCommitContent(MessageInputView.kt:207)
        at androidx.core.view.inputmethod.InputConnectionCompat$1.commitContent(InputConnectionCompat.java:275)
        at com.android.internal.view.IInputConnectionWrapper.executeMessage(IInputConnectionWrapper.java:566)
        at com.android.internal.view.IInputConnectionWrapper$MyHandler.handleMessage(IInputConnectionWrapper.java:89)
        at android.os.Handler.dispatchMessage(Handler.java:107)
        at android.os.Looper.loop(Looper.java:237)
        at android.app.ActivityThread.main(ActivityThread.java:8167)
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100) 

     Caused by: android.system.ErrnoException: open failed: ENOENT (No such file or directory)
        at libcore.io.Linux.open(Native Method)
        at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
        at libcore.io.BlockGuardOs.open(BlockGuardOs.java:252)
        at libcore.io.ForwardingOs.open(ForwardingOs.java:167)
        at android.app.ActivityThread$AndroidOs.open(ActivityThread.java:8044)
        at libcore.io.IoBridge.open(IoBridge.java:482)
        at java.io.FileOutputStream.<init>(FileOutputStream.java:235) 
        at java.io.FileOutputStream.<init>(FileOutputStream.java:186) 
        at com.getstream.sdk.chat.utils.StorageHelper.getCachedFileFromUri(StorageHelper.kt:41) 
        at com.getstream.sdk.chat.view.messageinput.MessageInputController.sendNormalMessage(MessageInputController.kt:100) 
        at com.getstream.sdk.chat.view.messageinput.MessageInputController.onSendMessageClick$stream_chat_android_release(MessageInputController.kt:87) 
        at com.getstream.sdk.chat.view.messageinput.MessageInputView.onSendMessage(MessageInputView.kt:281) 
        at com.getstream.sdk.chat.view.messageinput.MessageInputView.sendGifFromKeyboard(MessageInputView.kt:350) 
        at com.getstream.sdk.chat.view.messageinput.MessageInputView.access$sendGifFromKeyboard(MessageInputView.kt:47) 
        at com.getstream.sdk.chat.view.messageinput.MessageInputView$configInputEditText$3.onCommitContent(MessageInputView.kt:207) 
        at androidx.core.view.inputmethod.InputConnectionCompat$1.commitContent(InputConnectionCompat.java:275) 
        at com.android.internal.view.IInputConnectionWrapper.executeMessage(IInputConnectionWrapper.java:566) 
        at com.android.internal.view.IInputConnectionWrapper$MyHandler.handleMessage(IInputConnectionWrapper.java:89) 
        at android.os.Handler.dispatchMessage(Handler.java:107) 
        at android.os.Looper.loop(Looper.java:237) 
        at android.app.ActivityThread.main(ActivityThread.java:8167) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:496) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:1100) 

2021-03-13 11:49:24.512 E/Chat:Events: onFailure: java.io.EOFException
2021-03-13 11:49:24.514 E/Chat:Events: java.io.EOFException

        at okio.RealBufferedSource.require(RealBufferedSource.kt:199)
        at okio.RealBufferedSource.readByte(RealBufferedSource.kt:209)
        at okhttp3.internal.ws.WebSocketReader.readHeader(WebSocketReader.kt:119)
        at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.kt:102)
        at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.kt:293)
        at okhttp3.internal.ws.RealWebSocket$connect$1.onResponse(RealWebSocket.kt:195)
        at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)

2021-03-13 11:49:24.517 E/Chat:SocketService: Status code: -1, with stream code: 1003, description: See stack trace in logs. Intercept error in error handler of setUser
2021-03-13 11:49:24.518 E/Chat:SocketService: java.io.EOFException
        at okio.RealBufferedSource.require(RealBufferedSource.kt:199)
        at okio.RealBufferedSource.readByte(RealBufferedSource.kt:209)
        at okhttp3.internal.ws.WebSocketReader.readHeader(WebSocketReader.kt:119)
        at okhttp3.internal.ws.WebSocketReader.processNextFrame(WebSocketReader.kt:102)
        at okhttp3.internal.ws.RealWebSocket.loopReader(RealWebSocket.kt:293)
        at okhttp3.internal.ws.RealWebSocket$connect$1.onResponse(RealWebSocket.kt:195)
        at okhttp3.internal.connection.RealCall$AsyncCall.run(RealCall.kt:519)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
        at java.lang.Thread.run(Thread.java:919)
```

--- a/UNRELEASED_CHANGELOG.md
+++ b/UNRELEASED_CHANGELOG.md
@@ -16,6 +16,7 @@ Option `app:streamUiReactionsEnabled` in `MessageListView` to enable or disable 
 
 ## stream-chat-android
 ### 🐞 Fixed
+- Fixed crash when sending GIF from Samsung keyboard
 
 ### ⬆️ Improved
 

--- a/stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/model/AttachmentMetaData.kt
+++ b/stream-chat-android-ui-common/src/main/kotlin/com/getstream/sdk/chat/model/AttachmentMetaData.kt
@@ -49,4 +49,9 @@ public data class AttachmentMetaData(
             }
         }
     } ?: ModelType.attach_file
+
+    @InternalStreamChatApi
+    public companion object {
+        public const val DEFAULT_ATTACHMENT_TITLE: String = "attachment"
+    }
 }

--- a/stream-chat-android/src/main/java/com/getstream/sdk/chat/view/messageinput/MessageInputView.kt
+++ b/stream-chat-android/src/main/java/com/getstream/sdk/chat/view/messageinput/MessageInputView.kt
@@ -80,7 +80,7 @@ public class MessageInputView(context: Context, attrs: AttributeSet?) : Relative
         override fun sendToThread(
             parentMessage: Message,
             messageText: String,
-            alsoSendToChannel: Boolean
+            alsoSendToChannel: Boolean,
         ) {
             throw IllegalStateException("MessageInputView#messageSendHandler needs to be configured to send messages")
         }
@@ -89,7 +89,7 @@ public class MessageInputView(context: Context, attrs: AttributeSet?) : Relative
             parentMessage: Message,
             message: String,
             alsoSendToChannel: Boolean,
-            attachmentsFiles: List<File>
+            attachmentsFiles: List<File>,
         ) {
             throw IllegalStateException("MessageInputView#messageSendHandler needs to be configured to send messages")
         }
@@ -298,7 +298,7 @@ public class MessageInputView(context: Context, attrs: AttributeSet?) : Relative
         parentMessage: Message,
         message: String,
         alsoSendToChannel: Boolean,
-        attachmentFiles: List<File>
+        attachmentFiles: List<File>,
     ) {
         messageSendHandler.sendToThreadWithAttachments(
             parentMessage,
@@ -325,7 +325,7 @@ public class MessageInputView(context: Context, attrs: AttributeSet?) : Relative
 
     private fun sendGifFromKeyboard(
         inputContentInfo: InputContentInfoCompat,
-        flags: Int
+        flags: Int,
     ): Boolean {
         if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q &&
             flags and InputConnectionCompat.INPUT_CONTENT_GRANT_READ_URI_PERMISSION != 0
@@ -336,13 +336,14 @@ public class MessageInputView(context: Context, attrs: AttributeSet?) : Relative
                 return false
             }
         }
+        val title = inputContentInfo.contentUri.pathSegments.lastOrNull() ?: AttachmentMetaData.DEFAULT_ATTACHMENT_TITLE
         messageInputController.setSelectedAttachments(
             setOf(
                 AttachmentMetaData(
                     uri = inputContentInfo.contentUri,
                     type = ModelType.attach_image,
                     mimeType = ModelType.attach_mime_gif,
-                    title = inputContentInfo.description.label.toString(),
+                    title = title,
                 )
             )
         )
@@ -447,7 +448,7 @@ public class MessageInputView(context: Context, attrs: AttributeSet?) : Relative
             parentMessage: Message,
             message: String,
             alsoSendToChannel: Boolean,
-            attachmentsFiles: List<File>
+            attachmentsFiles: List<File>,
         )
 
         public fun editMessage(oldMessage: Message, newMessageText: String)

